ctags/json.go

```diff
@@ -119,19 +121,41 @@ func (lp *lockedParser) close() {
 
 // NewParser creates a parser that is implemented by the given
 // universal-ctags binary. The parser is safe for concurrent use.
-func NewParser(bin string) (Parser, error) {
-	if strings.Contains(bin, "universal-ctags") || strings.Contains(bin, "scip-ctags") {
-		opts := goctags.Options{
-			Bin: bin,
+func NewParser(parserType CTagsParserType, bin string) (Parser, error) {
+	if err := checkBinary(parserType, bin); err != nil {
+		return nil, err
```

Comment:
is this code path run often? ie. would this spawn a ton of processes?

Reply:
once per index job, so acceptable. IE before this change we would do one long running subprocess to universal-ctags (and communicate over stdio), now before we start the long lived process we run this first. Long lived == time it takes to index a repo.